pints/_mcmc/__init__.py

```diff
@@ -371,6 +398,7 @@ def __init__(
                 ok = False
             if not ok:
                 raise ValueError('Given method must extend pints.MCMCSampler.')
+        self._method = method
```

Comment:
Why this line?

Reply:
We use it to initialise each sampler in `run`.

Reply:
But you're already creating the samplers just below: `self._samplers = [method(x, sigma0)`

Reply:
Maybe create the samplers in `run()` ?

The "needs sensitivities" should probably be a static method so we can call it without having to create the samplers?

This relates to design questions about whether or not we want controllers to be single or multi use. It might actually make sense to let them be run more than once? Continue where you left off? That kind of thing. Separate issue, but doing this might be a step towards that!

Reply:
So replace

def needs_initial_phase(self):
    ...
    
with either

@staticmethod
def needs_initial_phase():
    ...
    
```
@staticmethod
def needs_initial_phase():
    ...
```

```
@classmethod
def needs_initial_phase(cls):
    ...
```